pkg/transportrequest/uploadcts_test.go

```diff
@@ -0,0 +1,156 @@
+package transportrequest
+
+import (
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestUploadCTS(t *testing.T) {
+
+	filesMock := mock.FilesMock{}
+	files = &filesMock
+	defer func() { files = piperutils.Files{} }()
+
+	t.Run("npm install command tests", func(t *testing.T) {
+		cmd := mock.ShellMockRunner{}
+		action := CTSUploadAction{
+			Connection:  CTSConnection{Endpoint: "", Client: "", User: "me", Password: "******"},
+			Application: CTSApplication{Pack: "", Name: "", Desc: ""},
+			Node: CTSNode{
+				DeployDependencies: []string{"@sap/my-dep"},
+				InstallOpts:        []string{"--verbose", "--registry", "https://registry.example.org"},
+			},
+			TransportRequestID: "12345678",
+			ConfigFile:         "ui5-deploy.yaml",
+			DeployUser:         "node",
+		}
+		err := action.Perform(&cmd)
+		if assert.NoError(t, err) {
+			assert.Contains(
+				t,
+				cmd.Calls[0],
+				"npm install --global --verbose --registry https://registry.example.org @sap/my-dep",
+			)
+			assert.Contains(
+				t,
+				cmd.Calls[0],
+				"su node",
+			)
+		}
+	})
+
+	t.Run("deploy command tests", func(t *testing.T) {
+		t.Run("all possible values provided", func(t *testing.T) {
+			cmd := mock.ShellMockRunner{}
+			action := CTSUploadAction{
+				Connection:  CTSConnection{Endpoint: "https://example.org:8080/cts", Client: "001", User: "me", Password: "******"},
+				Application: CTSApplication{Pack: "abapPackage", Name: "appName", Desc: "the Desc"},
+				Node: CTSNode{
+					DeployDependencies: []string{},
+					InstallOpts:        []string{},
+				},
+				TransportRequestID: "12345678",
+				ConfigFile:         "ui5-deploy.yaml",
+				DeployUser:         "doesNotMatterInThisCase",
+			}
+
+			err := action.Perform(&cmd)
+			if assert.NoError(t, err) {
+				assert.Contains(
+					t,
+					cmd.Calls[0],
+					"fiori deploy -f -y --username ABAP_USER --password ABAP_PASSWORD -e \"the Desc\" --noConfig -u https://example.org:8080/cts -l 001 -t 12345678 -p abapPackage -n appName",
+				)
+				assert.Equal(t, []string{"ABAP_USER=me", "ABAP_PASSWORD=******"}, cmd.Env)
+			}
+		})
+
+		t.Run("all possible values omitted", func(t *testing.T) {
```

Comment:
How about adding in the test description what the expected behaviour is? In this case it could be for example: "all possible values omitted, uses default values"

Reply:
Wouldn't "using default values" imply that there are default values used / contributed by the code unter test?
Here in this case we expect that the values are simply omitted in the command line and the, well "default values" are used from the config file - or even there are some default values inside the fiori deploy toolset which we use. IMO "using default values" leads to the wrong assumtion that the default values are provided by our code under test.

Reply:
ok